packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,291 @@
+  // event handlers
+
+  const handleColumnSelect = useCallback((columnId) => {
+    if (onColumnSelect) {
+      onColumnSelect(columnId);
+    }
+  }, [onColumnSelect]);
+
+  const onResizeMouseDown = useCallback((event, index, resizeColumnWidth) => {
+    // Store current table and column values for resize calculations
+    tableWidth.current = table.current.offsetWidth;
+    activeColumnPageX.current = event.pageX;
+    activeColumnWidth.current = resizeColumnWidth;
+
+    // Set the active index to the selected column
+    setActiveIndex(index);
+  }, []);
+
+  const onMouseMove = (event) => {
+    if (activeIndex == null) {
+      return;
+    }
+
+    // Ensure the new column width falls within the range of the minimum and maximum values
+    const diffX = event.pageX - activeColumnPageX.current;
+    const { minimumWidth, maximumWidth } = renderedColumns[activeIndex];
+    const newColumnWidth = Math.min(Math.max(activeColumnWidth.current + diffX, minimumWidth), maximumWidth);
+
+    // Update the width for the column in the state variable
+    const newColumns = [...renderedColumns];
+    newColumns[activeIndex].width = newColumnWidth;
+    setRenderedColumns(newColumns);
+
+    // Update the column and table width
+    table.current.style.width = `${tableWidth + (newColumnWidth - activeColumnWidth.current)}px`;
+  };
+
+  const onMouseUp = () => {
+    if (onColumnResize) {
+      onColumnResize(renderedColumns[activeIndex].id, renderedColumns[activeIndex].width);
+    }
+
+    // Remove active index
+    setActiveIndex(null);
+  };
+
+  // -------------------------------------
+
+  return (
+    <div>
+      <table
+        ref={tableResizeRef}
+        id={id}
+        aria-labelledby={ariaLabelledBy}
+        aria-label={ariaLabel}
+        className={cx('table', theme.className)}
+        {...(activeIndex != null && { onMouseUp, onMouseMove, onMouseLeave: onMouseUp })}
+      >
+        <ColumnContext.Provider
+          value={columnContextValue}
+        >
+          <ColumnHeader
+            columns={renderedColumns}
+            headerHeight={columnHeaderHeight}
+            tableHeight={tableHeight}
+            onResizeMouseDown={onResizeMouseDown}
+            onColumnSelect={handleColumnSelect}
+          />
+          <tbody>
+            {rows.map((row, index) => (
+              <Row
+                rowIndex={index + 1}
+                key={row.id}
+                height={rowHeight}
+                id={row.id}
+                cells={row.cells}
+                ariaLabel={row.ariaLabel}
+                displayedColumns={displayedColumns}
+                rowHeaderIndex={rowHeaderIndex}
+              />
+            ))}
+          </tbody>
+        </ColumnContext.Provider>
+      </table>
+      <VisuallyHiddenText aria-live="polite" aria-atomic="true" text={cellAriaLiveMessage} />
+    </div>
+  );
+}
+
+Table.propTypes = propTypes;
+Table.defaultProps = defaultProps;
+
+export default injectIntl(Table);
```

Comment:
I am wondering if we should memoize this component so that we don't have to worry about unnecessary rerenders.

Reply:
Did we consider this comment?